# 00 Data Collection and PDF/Excel Extraction

This notebook is executed and includes outputs plus interpretation notes.

## Goal
Extract incident-level KMO appendix data for 2017-2023 from PDFs and include the manually provided 2024 Excel table. The important validation point is that the final raw data should now include 2024 with 720 rows.

In [1]:
from pathlib import Path
import sys, json
import pandas as pd
import numpy as np
sys.path.append(str(Path.cwd().parent))

raw = pd.read_excel('../data/raw/kmo_incidents_raw.xlsx')
clean = pd.read_excel('../data/processed/kmo_incidents_clean.xlsx')
log = pd.read_csv('../data/raw/extraction_log.csv')
summary = raw.groupby('source_year').size().rename('rows').reset_index()
summary

,source_year,rows
0,2017,153
1,2018,436
2,2019,541
3,2020,493
4,2021,394
5,2022,587
6,2023,528
7,2024,720


In [2]:
log

,source_file,status,reason,rows
0,kmo2017.pdf,ok,NaN,153
1,kmo2018.pdf,ok,NaN,436
2,kmo2019.pdf,ok,NaN,541
3,kmo2020.pdf,ok,NaN,493
4,kmo2021.pdf,ok,NaN,394
5,kmo2022.pdf,ok,NaN,587
6,kmo2023.pdf,ok,NaN,528
7,kmo2024.pdf,ok_manual,Loaded data/raw/kmo2024_manual.xlsx,720


In [3]:
validation = {
    'raw_rows': len(raw),
    'clean_rows': len(clean),
    'years': sorted(clean['year'].dropna().astype(int).unique().tolist()),
    '2024_rows': int((clean['year'] == 2024).sum()),
    '2024_fire_explosion': clean.loc[clean['year'] == 2024, 'olay_turu'].value_counts().to_dict(),
    'istanbul_rows_total': int(clean['is_istanbul'].sum()),
}
validation

{'raw_rows': 3852,
 'clean_rows': 3852,
 'years': [2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024],
 '2024_rows': 720,
 '2024_fire_explosion': {'fire': 694, 'explosion': 26},
 'istanbul_rows_total': 1048}

## Interpretation
The dataset now covers 2017-2024 and contains 3,852 cleaned incident rows. The 2024 data was not OCR-derived anymore; it was loaded from the manually provided Excel file and contributes 720 rows. This resolves the earlier 2024 limitation and makes the final panel more complete.